src/Internal/Wallet/Cip30.js

```diff
@@ -5,16 +5,28 @@ exports._getNetworkId = conn => () => conn.getNetworkId();
 exports._getUtxos = maybe => conn => () =>
   conn.getUtxos().then(res => (res === null ? maybe.nothing : maybe.just(res)));
 
+const myGetCollateral = conn =>
+  conn.getCollateral !== undefined
+    ? conn.getCollateral
+    : conn.experimental.getCollateral;
+
 exports._getCollateral = maybe => conn => () =>
-  conn.experimental
-    .getCollateral()
+  myGetCollateral(conn)().then(utxos =>
+    utxos !== null && utxos.length ? maybe.just(utxos) : maybe.nothing
+  );
+
+// Yoroi wallet requires an explicit amount as argument to `getCollateral` and
```

Comment:
This is normal (according to CIP-30). We should do the same. If we don't, let's make this parameter required (as stated in https://cips.cardano.org/cips/cip30/#apigetcollateralparamsamountcborcoinpromisetransactionunspentoutputnull )

Reply:
Yes, but according to CIP 30 the argument should be `{ amount: ... }` which also fails with Yoroi.

Reply:
Ooof. Then let's keep it as is.